packages/core/src/config/config-reader.ts

```diff
@@ -47,37 +50,45 @@ export default class ConfigReader {
     return options;
   }
 
-  private loadConfigModule(): ((options: StrykerOptions) => void) | PartialStrykerOptions {
-    let configModule: PartialStrykerOptions | ((config: StrykerOptions) => void) = {};
+  private loadConfigModule(): PartialStrykerOptions {
+    const configExplorer = cosmiconfigSync(`${MODULE_NAME}`, {
+      searchPlaces: ['package.json', `${MODULE_NAME}.conf.js`, `${MODULE_NAME}.conf.json`],
+    });
+    let result: ReturnType<typeof configExplorer.search> = null;
 
     if (!this.cliOptions.configFile) {
+      this.log.debug('Searching for config file');
       try {
-        const configFile = require.resolve(path.resolve(`./${DEFAULT_CONFIG_FILE}`));
-        this.log.info(`Using ${path.basename(configFile)}`);
-        this.cliOptions.configFile = configFile;
+        result = configExplorer.search();
       } catch (e) {
-        this.log.info('No config file specified. Running with command line arguments.');
-        this.log.info('Use `stryker init` command to generate your config file.');
+        this.handleSyntaxError(e);
       }
-    }
 
-    if (typeof this.cliOptions.configFile === 'string') {
+      if (result) {
+        this.log.info(`Using ${result.filepath}`);
+        this.cliOptions.configFile = result.filepath;
+      } else {
+        this.log.info('No config file specified. Running with command line arguments only.');
+        this.log.info('Use `stryker init` command to generate your config file.');
+      }
+    } else {
       this.log.debug(`Loading config ${this.cliOptions.configFile}`);
-      const configFile = this.resolveConfigFile(this.cliOptions.configFile);
+      const resolvedFilename = this.resolveConfigFile(String(this.cliOptions.configFile));
```

Comment:
I'm not sure why you would like to use `String()` here since it wasn't used before? Any reasoning for this?